src/jiti.ts

```diff
@@ -302,15 +314,30 @@ export default function createJITI(
     }
 
     // Check for CJS cache
-    if (requiredModules[filename]) {
-      return _interopDefault(requiredModules[filename]?.exports);
+    if (cache[filename]) {
+      return _interopDefault(cache[filename]?.exports);
     }
     if (opts.requireCache && nativeRequire.cache[filename]) {
       return _interopDefault(nativeRequire.cache[filename]?.exports);
     }
 
     // Read source
-    let source = readFileSync(filename, "utf8");
+    const source = readFileSync(filename, "utf8");
+
+    // Evaluate module
+    return evalModule(source, { id, filename, ext, cache });
+  }
+
+  function evalModule(source: string, evalOptions: EvalModuleOptions = {}) {
+    // Resolve options
+    const id =
+      evalOptions.id ||
+      (evalOptions.filename
+        ? basename(evalOptions.filename)
+        : `_jitiEval.${evalOptions.ext || ".js"}`);
+    const filename = evalOptions.filename || _resolve(id);
+    const ext = evalOptions.ext || extname(filename);
+    const cache = (opts.cache || parentCache || {}) as ModuleCache;
```

Comment:
Because

```ts
const x: string = 'x';

(x || {}) as Record<string, any>
```

compiles, it would be much safer to type it like this:

```ts
const cache = (opts.cache as ModuleCache || parentCache || {} as ModuleCache);
```

or even better:

```ts
const cache: ModuleCache = opts.cache || parentCache || {};
```

This way it would error that `opts.cache` does not match `ModuleCache`.

Reply:
Do you mind to open a refactor PR?